File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def time_convert(t):
    year, month, day = map(int, t.split('.'))
    return year * 12 * 28 + 28 * month + day


def solution(today, terms, privacies):
    answer = []
    today = time_convert(today)
    month_dict = {}

    for term in terms:
        name, period = term.split()
        month_dict[name] = int(period) * 28

    for idx, privacies in enumerate(privacies):
        dates, name = privacies.split()
        total = time_convert(dates) + month_dict[name]
        if total <= today:
            answer.append(idx + 1)

    answer.sort()
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.cpp
#include <string>
#include <vector>
#include <sstream>
#include <map>
#include <iostream>
#include <algorithm>

using namespace std;

int time_convert(string time) {
    
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int> answer;
    map<char, int> tmap;
    
    int int_today = stoi(today.substr(0,4)) * 12 * 28 + stoi(today.substr(5,2)) * 28 + stoi(today.substr(8,2));
    
    for(int i=0;i<terms.size();i++){
        stringstream ss(terms[i]);
        char alpha; int month;
        ss>>alpha>>month;
        tmap[alpha] = month*28;
    }
    
    for(int i=0;i<privacies.size();i++){
        string year = privacies[i].substr(0,4);
        string month = privacies[i].substr(5,2);
        string day = privacies[i].substr(8,2);
        char alp = privacies[i].back();
        
        int temp = stoi(year) * 12 * 28 + stoi(month) * 28 + stoi(day);
        if (int_today >= temp + tmap[alp]) {
            answer.push_back(i+1);
        }
    }
    
    sort(answer.begin(), answer.end());
    
    return answer;
}